### PR TITLE
Implement SimpleNLPModel

### PR DESCRIPTION
**Pull request by abelsiqueira**
Includes consistency checks.

### CODE REVIEW

File: src/simple_model.jl [hunk 1]
```diff
@@ -0,0 +1,272 @@
...
+  nvar = length(x0)
+  length(lvar) == 0 && (lvar = -Inf*ones(nvar))
+  length(uvar) == 0 && (uvar =  Inf*ones(nvar))
+  ncon = maximum([length(lcon); length(ucon); length(y0)])
+  if nnzh == 0
+    try
+      if ncon == 0
+        A = hess(x0)
+      else
+        A = hess(x0, y0)
```
Comment (dpo) [inline src/simple_model.jl]: Shouldn't you check that `y0` was specified here, and possible set it to a vector that's likely to bring out all constraint contributions? I would choose something like `y0 = [1, -1, 1, -1, ...]`.

Comment (abelsiqueira) [inline src/simple_model.jl]: True. I think I'm gonna create a fake y0 even if y0 is specified (it could be zeros).

**Comment by abelsiqueira** [comment]
Updated

**Comment by dpo** [comment]
Great, thanks!

**Pull request merged**